From 729e877c64869bac4c66926ee5f52b3509a4765b Mon Sep 17 00:00:00 2001
From: Alex Porosanu <alexandru.porosanu@freescale.com>
Date: Fri, 28 Nov 2014 10:50:40 +0200
Subject: [PATCH 083/399] dpa_offload: sync with CAAM changes

The CAAM driver function caam_jr_strstatus() has changed
its usage, and now it performs the interpretation and
printing of the error itself. This patch syncs dpa_offload's
sources with the updated version.

Change-Id: I5b7b04386dbc03c11e3277adb77af985cc6418ca
Signed-off-by: Alex Porosanu <alexandru.porosanu@freescale.com>
Reviewed-on: http://git.am.freescale.net:8181/24679
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Cristian Stoica <cristian.stoica@freescale.com>
Reviewed-by: Marian-Cornel Chereji <marian.chereji@freescale.com>
Reviewed-by: Richard Schmitt <richard.schmitt@freescale.com>
[Lu:Original patch taken from
QorIQ-SDK-V1.7-SOURCE-20141218-yocto.iso]
Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 drivers/staging/fsl_dpa_offload/dpa_ipsec_desc.c |    7 +++----
 1 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dpa_ipsec_desc.c b/drivers/staging/fsl_dpa_offload/dpa_ipsec_desc.c
index 6a280e5..ae767d8 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_ipsec_desc.c
+++ b/drivers/staging/fsl_dpa_offload/dpa_ipsec_desc.c
@@ -1935,10 +1935,9 @@ static void split_key_done(struct device *dev, u32 *desc, u32 err,
 {
 	register atomic_t *done = context;
 
-	if (err) {
-		char tmp[256];
-		dev_err(dev, "%s\n", caam_jr_strstatus(tmp, err));
-	}
+	if (err)
+		caam_jr_strstatus(dev, err);
+
 	atomic_set(done, 1);
 }
 
-- 
1.7.5.4

